### ISSUE TITLE
Cannot send email

### ISSUE THREAD

**Issue by Valinalo** [jmapio/jmap-perl#30]
When I try send email from Drafts to Outbox:

`[
    [ "setMessages", {
"update":
{
"md5a07f187":{
"mailboxIds":["5"]
}
}
}, "#1" ]
]`

, I get error:

`[
    [
        "error",
        {
            "message": "isa check for \"helo\" failed: undef is not a string! at (eval 310) line 108.\n\teval {...} called at (eval 310) line 107\n\tEmail::Sender::Transport::SMTPS::new(\"Email::Sender::Transport::SMTPS\", HASH(0x2bee918)) called at /home/jmap/jmap-perl/JMAP/Sync/Standard.pm line 103\n\tJMAP::Sync::Standard::send_email(JMAP::Sync::Standard=HASH(0x66c9b00), \"From: bob\\@bicsh.com\\x{d}\\x{a}To: alice\\@bicsh.com\\x{d}\\x{a}Cc: \\x{d}\\x{a}Bcc: \\x{d}\\x{a}Subjec\"...) called at /home/jmap/jmap-perl/JMAP/ImapDB.pm line 135\n\tJMAP::ImapDB::backend_cmd(JMAP::ImapDB=HASH(0x5969338), \"send_email\", \"From: bob\\@bicsh.com\\x{d}\\x{a}To: alice\\@bicsh.com\\x{d}\\x{a}Cc: \\x{d}\\x{a}Bcc: \\x{d}\\x{a}Subjec\"...) called at /home/jmap/jmap-perl/JMAP/ImapDB.pm line 1109\n\tJMAP::ImapDB::update_messages(JMAP::ImapDB=HASH(0x5969338), HASH(0x23b21c8), CODE(0x66f0160)) called at /home/jmap/jmap-perl/JMAP/API.pm line 1178\n\tJMAP::API::setMessages(JMAP::API=HASH(0x66c9a28), HASH(0x23b21f8), \"#1\") called at /home/jmap/jmap-perl/bin/apiendpoint.pl line 601\n\teval {...} called at /home/jmap/jmap-perl/bin/apiendpoint.pl line 601\n\tJMAP::Backend::handle_jmap(JMAP::ImapDB=HASH(0x5969338), ARRAY(0x23be420), \"#0\") called at /home/jmap/jmap-perl/bin/apiendpoint.pl line 201\n\teval {...} called at /home/jmap/jmap-perl/bin/apiendpoint.pl line 187\n\tJMAP::Backend::__ANON__(AnyEvent::Handle=HASH(0x6489048), ARRAY(0x23eb1f0)) called at /usr/lib/x86_64-linux-gnu/perl5/5.20/AnyEvent/Handle.pm line 1735\n\tAnyEvent::Handle::__ANON__(AnyEvent::Handle=HASH(0x6489048)) called at /usr/lib/x86_64-linux-gnu/perl5/5.20/AnyEvent/Handle.pm line 1310\n\tAnyEvent::Handle::_drain_rbuf(AnyEvent::Handle=HASH(0x6489048)) called at /usr/lib/x86_64-linux-gnu/perl5/5.20/AnyEvent/Handle.pm line 2009\n\tAnyEvent::Handle::__ANON__(EV::IO=SCALAR(0x23b2168), 1) called at /home/jmap/jmap-perl/bin/apiendpoint.pl line 120\n\teval {...} called at /home/jmap/jmap-perl/bin/apiendpoint.pl line 120\n\tJMAP::Backend::process_request(JMAP::Backend=HASH(0x1d76a60), Net::Server::Proto::TCP=GLOB(0x63d0060)) called at /usr/local/share/perl/5.20.2/Net/Server.pm line 74\n\tNet::Server::run_client_connection(JMAP::Backend=HASH(0x1d76a60)) called at /usr/local/share/perl/5.20.2/Net/Server/Fork.pm line 199\n\tNet::Server::Fork::run_client_connection(JMAP::Backend=HASH(0x1d76a60)) called at /usr/local/share/perl/5.20.2/Net/Server/Fork.pm line 144\n\tNet::Server::Fork::loop(JMAP::Backend=HASH(0x1d76a60)) called at /usr/local/share/perl/5.20.2/Net/Server.pm line 60\n\tNet::Server::run(\"JMAP::Backend\", \"host\", \"127.0.0.1\", \"port\", 5000) called at /home/jmap/jmap-perl/bin/apiendpoint.pl line 124\n",
            "type": "serverError"
        },
        "#1"
    ]
]`

When I connect to this SMTP server with different client, it works. Can somebody help me, where can be problem.

**Comment by brong** [comment]
Coming to this a long time later!

Email::Sender::Transport::SMTPS::new... so either authentication or SSL negotiation was failing.

Looking at the changes on that module:

0.04  2017-04-05 09:20:56 CST
        - DEPRECATED in favour of Email::Sender::Transport::SMTP

So it looks like maybe I should port us over to a new version Transport::SMTP.  I'll get you to test again when that's done if you're still interested.  That's a really crappy error message we got!

**Comment by brong** [comment]
Hmm, on my brand new attempt on the latest I got this:

JMAP CMD setMessages at bin/apiendpoint.pl line 599.
DBD::SQLite::db do failed: jmailboxes.jmailboxid may not be NULL at /home/jmap/jmap-perl/JMAP/DB.pm line 830.

Which I guess means I screwed up something in mailbox generation.  There's a whole new sending process being written up soon though, so I'm tempted to just wait for it:

Follow the long thread from here:

https://mailarchive.ietf.org/arch/msg/jmap/ZiXC-x-lAchRc2tc8SfOK2L_QJg

If you want to see the full discussion about different sending methods which are being covered.
